### ISSUE TITLE
Sample apiserver crashes with Failed to list *v1.ValidatingWebhookConfiguration

### ISSUE THREAD

**Issue by raushan2016** [kubernetes/kubernetes#84407]
**What happened**:
I1026 21:48:49.499318       1 dynamic_cafile_content.go:166] Starting request-header::/tmp/requestheader-client-ca-file714625551
I1026 21:48:49.499683       1 dynamic_serving_content.go:129] Starting serving-cert::/apiserver.local.config/certificates/tls.crt::/apiserver.local.config/certificates/tls.key
I1026 21:48:49.500181       1 secure_serving.go:160] Serving securely on [::]:443
I1026 21:48:49.500199       1 tlsconfig.go:220] Starting DynamicServingCertificateController
E1026 21:48:49.594139       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.0.0-20191025225928-af0e48646694/tools/cache/reflector.go:105: Failed to list *v1.ValidatingWebhookConfiguration: the server could not find the requested resource
E1026 21:48:49.594226       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.0.0-20191025225928-af0e48646694/tools/cache/reflector.go:105: Failed to list *v1.MutatingWebhookConfiguration: the server could not find the requested resource
E1026 21:48:50.596203       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.0.0-20191025225928-af0e48646694/tools/cache/reflector.go:105: Failed to list *v1.ValidatingWebhookConfiguration: the server could not find the requested resource
E1026 21:48:50.597214       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.0.0-20191025225928-af0e48646694/tools/cache/reflector.go:105: Failed to list *v1.MutatingWebhookConfiguration: the server could not find the requested resource
E1026 21:48:51.598371       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.0.0-20191025225928-af0e48646694/tools/cache/reflector.go:105: Failed to list *v1.ValidatingWebhookConfiguration: the server could not find the requested resource
E1026 21:48:51.598933       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.0.0-20191025225928-af0e48646694/tools/cache/reflector.go:105: Failed to list *v1.MutatingWebhookConfiguration: the server could not find the requested resource

**What you expected to happen**:
sample-apiserver should start correctly

**How to reproduce it (as minimally and precisely as possible)**:
Deploy the sample-apiserver from master
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.11", GitCommit:"25074a190ef2a07d8b0ed38734f2cb373edfb868", GitTreeState:"clean", BuildDate:"2019-09-18T14:34:46Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration: Azure
- OS (e.g: `cat /etc/os-release`):
NAME="Ubuntu"
VERSION="18.04.2 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18
- Kernel (e.g. `uname -a`):
Linux mymachinedev 5.0.0-1023-azure #24~18.04.1-Ubuntu SMP Wed Oct 2 20:23:29 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by raushan2016** [comment]
@kubernetes/sig-api-machinery-bugs

**Comment by raushan2016** [comment]
Seems like the cluster has 
API Version:  admissionregistration.k8s.io/v1beta1
Kind:         MutatingWebhookConfiguration

But the sample api-server expects 
API Version:  admissionregistration.k8s.io/v1
Kind:         MutatingWebhookConfiguration

Is there a way to make it work by using the last good version of apiserver which works with v1beta1

**Comment by liggitt** [comment]
The v1.16.0 level of the sample apiserver still uses v1beta1

**Comment by liggitt** [comment]
The sample apiserver in master (and in 1.17) will list v1 admission webhooks.

/close

**Comment by codyaray** [comment]
I'm running into the issue while running `sample-apiserver:1.17`. It looks like only `admissionregistration.k8s.io/v1beta1` is supported here. (Very new to this, so let me know if I'm missing something or looking at the wrong bits).

```
$ kubectl -n wardle get deploy wardle-server -o yaml | grep image:
        image: gcr.io/kubernetes-e2e-test-images/sample-apiserver:1.17
      - image: quay.io/coreos/etcd:v3.4.7

$ curl localhost:8080/apis/admissionregistration.k8s.io
{
  "kind": "APIGroup",
  "apiVersion": "v1",
  "name": "admissionregistration.k8s.io",
  "versions": [
    {
      "groupVersion": "admissionregistration.k8s.io/v1beta1",
      "version": "v1beta1"
    }
  ],
  "preferredVersion": {
    "groupVersion": "admissionregistration.k8s.io/v1beta1",
    "version": "v1beta1"
  }
}
```

My logs have the same errors as the OP:
```
E0420 14:24:31.890558       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.17.0/tools/cache/reflector.go:108: Failed to list *v1.MutatingWebhookConfiguration: the server could not find the requested resource
E0420 14:24:32.837903       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.17.0/tools/cache/reflector.go:108: Failed to list *v1.ValidatingWebhookConfiguration: the server could not find the requested resource
```

**Comment by liggitt** [comment]
> It looks like only `admissionregistration.k8s.io/v1beta1` is supported here.

What version kube-apiserver are you running against? This API is only available in kubernetes 1.16+

**Comment by rpherrera** [comment]
https://github.com/kubernetes/kube-state-metrics/issues/1090#issuecomment-728698307

Repeating myself:

I was having this issue with a cluster on `v1.15.12-gke.20` version.

The most recent version of kube-state-metrics that worked for me without flooding the logs with these errors was: `v1.9.4`
